src/filters/dark_filter.c

```diff
@@ -3,62 +3,56 @@
 #define MAX_COLOR 255
 #define DARK_FACTOR 0.4
 
-int dark_filter(const char *inputFile, const char *outputFile)
+int dark_filter(const char *input_file, const char *output_file)
 {
-    FILE *fileIn = fopen(inputFile, "rb");
-    FILE *fileOut = fopen(outputFile, "wb+");
-    unsigned char headerInfo[54];
+    FILE *file_in = fopen(input_file, "rb");
+    FILE *file_out = fopen(output_file, "wb+");
+    unsigned char header_info[54];
     int i, r, g, b;
 
-    if (fileIn == NULL || fileOut == NULL)
+    if (file_in == NULL || file_out == NULL)
     {
         printf("File does not exist.\n");
-        if (fileIn != NULL)
-            fclose(fileIn);
-        if (fileOut != NULL)
-            fclose(fileOut);
+        if (file_in != NULL)
+            fclose(file_in);
+        if (file_out != NULL)
+            fclose(file_out);
         return 1;
     }
-
     // read header info of image
     for (i = 0; i < 54; i++)
     {
-        headerInfo[i] = getc(fileIn);
+        header_info[i] = getc(file_in);
     }
-
     // write header info to output file
-    fwrite(headerInfo, sizeof(unsigned char), 54, fileOut);
+    fwrite(header_info, sizeof(unsigned char), 54, file_out);
     // extract attributes from image header
-    int height = *(int *)&headerInfo[18];
-    int width = *(int *)&headerInfo[22];
-    int size = height * width;
-
-    unsigned char buffer[size][3];
-
+    int height = *(int *)&header_info[18];
+    int width = *(int *)&header_info[22];
+    int pixels_in_image = height * width;
+    // check if image has a color table
+    unsigned char buffer[pixels_in_image][3];
```

Comment:
https://github.com/VerzatileDevOrg/C_image_processing/pull/8/files#r1301252661

Reply:
Changed so it is inline with other filter files.